components/switch/src/Switch.js

```diff
@@ -62,14 +48,125 @@ const asArray = val => (Array.isArray(val) ? val : [val]);
 // ==============================
 // Component
 // ==============================
+export const Switch = ({
+	name,
+	label,
+	checked: isChecked,
+	onChange,
+	size,
+	block,
+	flipped,
+	toggleText,
+	disabled,
+	srOnlyText,
+	...props
+}) => {
+	const { COLORS, [pkg.name]: overwritesWithTokens } = useBrand();
+	const mq = useMediaQuery();
+	const [checked, setChecked] = useState(isChecked);
+	const sizeArr = responsiveMap(asArray(size));
 
-export const Switch = ({ size, ...props }) => {
-	const flexiSize = responsiveMap(asArray(size));
+	const overwrites = {
+		toggleCSS: {},
+		toggleTextCSS: {},
+		CSS: {},
+		Label,
+	};
+
+	merge(overwrites, overwritesWithTokens);
+
+	useEffect(() => {
+		setChecked(isChecked);
+	}, [isChecked]);
+
+	const handleChange = () => wrapHandlers(onChange, () => setChecked(!checked));
```

Comment:
@dominikwilkowski I'm unsure if this is the correct use of wrapHandlers. From looking at the wrap handlers code, we run both the passed in consumerHandler and our own internal handler unless specified in the consumerHandler via preventDefault() ?

Reply:
That looks absolutely correct :)
For more infos look at https://github.com/WestpacGEL/GEL/blob/develop/components/core/examples/90-wrapHandlers.js?ts=2